### ISSUE TITLE
Question about dataset splitting and data leakage

### ISSUE THREAD

**Issue by ChangxiChi** [theislab/chemCPA#174]
Brilliant work!

In the main experiments of the paper, the evaluation focuses on "unseen drug-covariate combinations". However, in the data splitting implemented in your preprocessing folder, the splits do not hold out all cells under certain unseen drug-covariate combinations as the test set. Instead, only a portion is assigned to the test set, which may lead to data leakage.

Could you please point me to the specific code that generates a dataset split where certain drug-covariate combinations are completely held out from the training set?
